src/client/src/Room.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+
+export const Room = ({ room, setRoom, setMessages }) => {
+  const [rooms, setRooms] = useState(['room1', 'room2', 'room3']);
+
+  const handleRoomChange = (event) => {
+    setRoom(event.target.value);
+  };
+
+  const handleCreateRoom = () => {
+    const roomName = prompt('Enter the name for the new room:');
+    if (roomName) {
+      setRooms([...rooms, roomName]);
+    }
+  };
+
+  const handleRenameRoom = () => {
+    const newRoomName = prompt('Enter the new name for the room:', room);
+    if (newRoomName) {
+      setMessages(prevMessages => {
+        return prevMessages.map(message => {
+          if (message.room === room) {
+            message.room = newRoomName;
```

Comment:
You are directly modifying the state. In React, you should never mutate the state directly. Instead, you should return a new state object with the desired changes.